src/FSharp.Core/reflect.fs

```diff
@@ -373,17 +373,7 @@ module internal Impl =
     // UNION DECOMPILATION
 
     // Get the type where the type definitions are stored
-    let getUnionCasesTyp (typ: Type, _bindingFlags) =
-#if CASES_IN_NESTED_CLASS
-        let casesTyp = typ.GetNestedType("Cases", bindingFlags)
-
-        if casesTyp.IsGenericTypeDefinition then
-            casesTyp.MakeGenericType(typ.GetGenericArguments())
-        else
-            casesTyp
-#else
-        typ
-#endif
+    let getUnionCasesTyp (typ: Type, _bindingFlags) = typ
```

Comment:
I think we can remove this whole function and just `typ` whenever this is called

Reply:
I decided not to do that now, just remove the ifdefs. It will be inlined anyway.